src/Model/ChangeStreamIterator.php

```diff
@@ -244,9 +246,19 @@ private function extractResumeToken($document)
             return $this->extractResumeToken($document->bsonSerialize());
         }
 
-        $resumeToken = is_array($document)
-            ? ($document['_id'] ?? null)
-            : ($document->_id ?? null);
+        if ($document instanceof Document) {
```

Comment:
This is to support `watch` operations with a codec supplied, so we can avoid messing with the type map on every iteration.